src/server/scripts/Northrend/FrozenHalls/HallsOfReflection/halls_of_reflection.cpp

```diff
@@ -219,49 +252,74 @@ class npc_hor_leader : public CreatureScript
-                        if (Creature* c = pInstance->instance->GetCreature(pInstance->GetGuidData(NPC_DARK_RANGER_LORALEN)))
-                            c->SetVisible(true);
+                        
                     }
                     break;
                 case EVENT_PRE_INTRO_2:
                     if (me->GetEntry() == NPC_JAINA_PART1)
-                        Talk(SAY_JAINA_INTRO_1);
-                    me->GetMotionMaster()->MovePoint(0, SpawnPos);
+                       Talk(SAY_JAINA_INTRO_1);                 
+                    me->GetMotionMaster()->MovePoint(0, SpawnPos);                   
+                    break;
+                case EVENT_TALK_LEADER_1:
+                    me->SetSheath(SHEATH_STATE_UNARMED);
+                    Talk(me->GetEntry() == NPC_JAINA_PART1 ? SAY_JAINA_INTRO_2 : SAY_SYLVANAS_INTRO_1);                  
+                    break;
+                case EVENT_EMOTE_LEADER_1:
+                    me->HandleEmoteCommand(EMOTE_ONESHOT_EXCLAMATION);
                     break;
                 case EVENT_PRE_INTRO_3:
                     me->SetNpcFlag(UNIT_NPC_FLAG_GOSSIP | UNIT_NPC_FLAG_QUESTGIVER);
-                    Talk(me->GetEntry() == NPC_JAINA_PART1 ? SAY_JAINA_INTRO_2 : SAY_SYLVANAS_INTRO_1);
-                    me->SetFacingTo(0.89f);
-                    break;
-
+                    me->SetSheath(SHEATH_STATE_MELEE);
+                    break;                  
                 case EVENT_START_INTRO:
-                    shortver = false;
+                    shortver = false;                   
                     me->SetUInt32Value(UNIT_NPC_EMOTESTATE, EMOTE_ONESHOT_NONE);
                     me->GetMotionMaster()->MovePoint(0, MoveThronePos);
-                    if (Creature* c = pInstance->instance->GetCreature(pInstance->GetGuidData(NPC_DARK_RANGER_LORALEN)))
-                        c->GetMotionMaster()->MovePoint(0, LoralenFollowPos);
+                    if (Creature* pLoralen = pInstance->instance->GetCreature(pInstance->GetGuidData(NPC_DARK_RANGER_LORALEN)))
+                    {
+                        pLoralen->LoadEquipment(true);
+                        pLoralen->SetVisible(true);
+                        pLoralen->SetWalk(true);
+                        pLoralen->GetMotionMaster()->MovePoint(0, LoralenMidleFollowPos);
+                        
+                    }
                     // Begining of intro is differents between factions as the speech sequence and timers are differents.
                     if (me->GetEntry() == NPC_JAINA_PART1)
                         events.ScheduleEvent(EVENT_INTRO_A2_1, 10s);
                     else
-                        events.ScheduleEvent(EVENT_INTRO_H2_2, 10s);
+                        events.ScheduleEvent(EVENT_INTRO_H2_2, 9s);
+                        events.ScheduleEvent(EVENT_LORALEN_MOVE_1, 24s);
+                        events.ScheduleEvent(EVENT_LORALEN_MOVE_2, 32s);
```

Comment:
The above two scheduled events are not being executed in the else scope, you need to be explicit with your scope and use curly brackets. 

```diff
else
+{
    events.ScheduleEvent(EVENT_INTRO_H2_2, 9s);
    events.ScheduleEvent(EVENT_LORALEN_MOVE_1, 24s);
    events.ScheduleEvent(EVENT_LORALEN_MOVE_2, 32s);
+}
```

If the events are not intended to be executed in the else scope, remove one level of indentation from those above two statements.